pkg/adapter/kubernetesevents/adapter.go

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubernetesevents
+
+import (
+	"context"
+	"github.com/knative/pkg/logging"
+	"github.com/knative/pkg/signals"
+	"go.uber.org/zap"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/knative/pkg/cloudevents"
+	corev1 "k8s.io/api/core/v1"
+	coreinformers "k8s.io/client-go/informers/core/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const (
+	eventType = "dev.knative.k8s.event"
+)
+
+type Adapter struct {
+	// Namespace is the kubernetes namespace to watch for events from.
+	Namespace string
+	// SinkURI is the URI messages will be forwarded on to.
+	SinkURI string
+
+	ctx context.Context
```

Comment:
This context is stored and used for the lifetime of the process, which seems like an invalid use of context. It might be simpler and more idiomatic to give the struct a Logger field, or alternately add a `NewAdapter` function that takes a logger and sets an unexported `logger` field using some transformation (`WithName`, etc).